app/pages/project/disks/DisksPage.tsx

```diff
@@ -33,22 +40,22 @@ import { pb } from 'app/util/path-builder'
 
 import { fancifyStates } from '../instances/instance/tabs/common'
 
-function AttachedInstance({
-  instanceId,
-  ...projectSelector
-}: {
-  project: string
-  instanceId: string
-}) {
-  const { data: instance } = useApiQuery('instanceView', {
-    path: { instance: instanceId },
-  })
-
-  const instanceLinkCell = linkCell((instanceName) =>
-    pb.instancePage({ ...projectSelector, instance: instanceName })
+function InstanceNameFromId({ value: instanceId }: { value: string | null }) {
+  const { project } = useProjectSelector()
+  const { data: instance } = useApiQuery(
+    'instanceView',
+    { path: { instance: instanceId! } },
+    { enabled: !!instanceId }
   )
 
-  return instance ? instanceLinkCell({ value: instance.name }) : null
+  if (!instanceId) return null
+  if (!instance) return <SkeletonCell />
+
+  return (
+    <LinkCell to={pb.instancePage({ project, instance: instance.name })}>
+      {instance.name}
+    </LinkCell>
+  )
```

Comment:
this is unrelated but I made all the NameForId components use LinkCell and have the nice Skeleton state